### DISCUSSION TITLE
twitter.com - Subscribe to Premium, french language

### DISCUSSION THREAD

**Discussion by M0lyk** [uBlockOrigin/uAssets#22061] [Q&A] [answered]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin (uBO) along with other content blocker extensions.
- [X] I have verified that the web browser's built-in blocker or [DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking) (standalone or through a VPN) is not causing the issue.
- [ ] I did not answer truthfully to **ALL** the above checkboxes.

### URL(s) where the issue occurs.

```yaml
https://twitter.com/home
```

### Description

When I'm on the Twitter homepage, there's always an insert in the top right-hand corner suggesting that I sign up for a Premium subscription.

FR = "Abonnez‑vous à Premium
Abonnez‑vous pour profiter de nouvelles fonctionnalités et recevoir une part des revenus publicitaires si vous respectez les critères."

EN = "Subscribe to Premium
Subscribe to enjoy new features and receive a share of advertising revenue if you meet the criteria."

I'd like to remove this insert. Thank you to help me.

---------
This is my first post here and I hope I've followed all the rules, I've followed them in good faith, please excuse me if I've done anything wrong.

### Other extensions used

none for testing

### Screenshot(s)

<details>
<summary>Screenshot(s)</summary>

![2024-01-18 13_36_31-Accueil _ X — Mozilla Firefox](https://github.com/uBlockOrigin/uAssets/assets/1733940/1d15eeeb-6a2f-484e-b5ab-9a5d342c2021)

</details>

### Configuration

<details>
<summary>Details</summary>

```yaml
uBlock Origin: 1.55.0
Firefox: 121
filterset (summary):
 network: 35951
 cosmetic: 18916
 scriptlet: 18803
 html: 2016
listset (total-discarded, last-updated):
 removed:
  easylist: null
  easyprivacy: null
  urlhaus-1: null
  plowe-0: null
 default:
  user-filters: 917-2, never
  ublock-filters: 37072-184, 1h.15m Δ
  ublock-badware: 7746-11, 1h.15m Δ
  ublock-privacy: 969-0, 1h.15m Δ
  ublock-unbreak: 2236-1, 1h.15m Δ
  ublock-quick-fixes: 229-0, 1h.15m Δ
  FRA-0: 26863-69, 21m
filterset (user): [array of 916 redacted]
trustedset:
 added: [array of 595 redacted]
 removed:
  about-scheme
  chrome-extension-scheme
  chrome-scheme
  edge-scheme
switchRuleset:
 added: [array of 5 redacted]
hostRuleset:
 added: [array of 3974 redacted]
userSettings:
 advancedUserEnabled: true
 cloudStorageEnabled: true
 webrtcIPAddressHidden: true
hiddenSettings:
 cloudStorageCompression: false
supportStats:
 allReadyAfter: 2227 ms (selfie)
 maxAssetCacheWait: 607 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 3
 network:
  twitter.com: 3
 extended:
  ##[data-testid="primaryColumn"] [data-testid="cellInnerDiv"] > d…

```
</details>

**Comment by gorhill** [comment]
Moved to Discussions, this is not an issue to be fixed in filter lists.

**Reply by M0lyk** [reply]
Hello gorhill,

Yes, I would need a filter, I didn't want to report a bug. Thanks

**Reply by nalllllllaaaaaa** [reply]
hey

**Comment by gorhill** [comment]
Does this work for you?

    twitter.com##[data-testid="sidebarColumn"] aside[aria-label="Subscribe to Premium"]

If not, replace `Subscribe to Premium` with `Abonnez-vous à Premium`.

**Reply by M0lyk** [reply]
So, I'm using Firefox v121 on Windows 11.
+ To test, I once again disabled all my extensions except uBlock Origin.
+ default DNS, no VPN.
+ In uBlock, I disabled all optional and personal lists, and tested again.
+ Then I re-enabled my personal list, making sure there were no more items starting with twitter.com.
+ After that, I ran the 2 suggested tests, and there are no results.

![2024-01-18 15_33_31-](https://github.com/uBlockOrigin/uAssets/assets/1733940/0977601a-1878-47d0-ae7c-e2ac331de469)

**Reply by Yuki2718** [reply]
So the rule worked or not?

**Reply by Yuki2718** [reply, marked as answer]
Language-independent alternative: `twitter.com##div[data-testid="sidebarColumn"] aside[role="complementary"]:has(> a[href^="/i/verified-choose"])`

**Reply by M0lyk** [reply]
no results -> no effect, didn't work

**Reply by Yuki2718** [reply]
The rule should work as long as the language matches.

**Comment by M0lyk** [comment]
> Language-independent alternative: `twitter.com##div[data-testid="sidebarColumn"] aside[role="complementary"]:has(> a[href^="/i/verified-choose"])`

^ This is working. thank you very much !!
